0_intro_to_jax_for_ml.ipynb

```diff
@@ -0,0 +1,2081 @@
+{
```

Comment:
<p>One general comment: it can be very useful for people coming from other frameworks (most notably PyTorch) to get some idea of what problem JAX is trying to solve. Why was JAX created in a world that already has PyTorch, and why should they be interested in learning about it in these tutorials? I can imagine this coming up a lot. (this could go at the end of the tutorial for example). </p><p> </p><blockquote>JAX is <strong>not</strong> a replacement for Pytorch or Tensorflow, but a lower-level library commonly used with higher-level neural network libraries such as <a href="https://github.com/deepmind/dm-haiku" target="_blank">Haiku</a> or <a href="https://github.com/google/flax" target="_blank">Flax</a>.</blockquote><p>This is a confusing statement. JAX is 100% intended to be a replacement for PyTorch and TensorFlow. It can be used in its plain form, but it is often used via libraries that make lots of things simpler (precisely as TensorFlow was always used via Keras, TensorFlow Slim, etc etc).</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
@sbodenstein Thanks for comments. 

On "why JAX" - I guess I was trying to answer this in the https://colab.research.google.com/github/deep-learning-indaba/indaba-pracs-2022/blob/prac1-intro-ml-using-jax/0_intro_to_ml_using_jax.ipynb#scrollTo=TSj972IWxTo2 section (JAX is backed agnostic and all the useful transforms), but I get what you are saying. So you think having this in the intro or conclusion would work best? Like a "Why JAX" section. 

On the "JAX is not a replacement ... " part, I have updated this (https://colab.research.google.com/github/deep-learning-indaba/indaba-pracs-2022/blob/prac1-intro-ml-using-jax/0_intro_to_ml_using_jax.ipynb#scrollTo=Enx0WUr8tIPf) . Please let me know if you this what you think :)